packages/client/src/devices/InputMediaDeviceManagerState.ts

```diff
@@ -43,10 +44,47 @@ export abstract class InputMediaDeviceManagerState<C = MediaTrackConstraints> {
    */
   defaultConstraints$ = this.defaultConstraintsSubject.asObservable();
 
+  /**
+   * An observable that will emit `true` if browser/system permission
+   * is granted, `false` otherwise.
+   */
+  hasBrowserPermission$ = new Observable<boolean>((subscriber) => {
+    const notifyGranted = () => subscriber.next(true);
+    if (isReactNative() || !this.permissionName) return notifyGranted();
+
+    let permissionState: PermissionStatus;
+    const notify = () => subscriber.next(permissionState.state === 'granted');
+    navigator.permissions
+      .query({ name: this.permissionName })
+      .then((permissionStatus) => {
+        permissionState = permissionStatus;
+        permissionState.addEventListener('change', notify);
+        notify();
+      })
+      .catch(() => {
+        // permission doesn't exist or can't be queried -> assume it's granted
+        // an example would be Firefox,
+        // where neither camera microphone permission can be queried
+        notifyGranted();
+      });
+
+    return () => {
+      permissionState?.removeEventListener('change', notify);
+    };
+  });
```

Comment:
I think we can use the `shareReplay(1)` pipe, and then the source method (the parameter of the constructor), will only run once, regardless of the number of subscribers

Reply:
Good point, I'll update the implementation accordingly.

Reply:
Going with `shareReplay({ bufferSize: 1, refCount: true })` as otherwise, the "observable cleanup" never runs once others unsubscribe.